community/common/src/main/java/org/neo4j/kernel/lifecycle/LifecycleException.java

```diff
@@ -27,42 +27,63 @@
+        StringBuilder message = new StringBuilder();
+        switch ( to )
         {
             case STOPPED:
-                if(from == LifecycleStatus.NONE)
+                if ( from == LifecycleStatus.NONE )
                 {
-                    return String.format( "Component '%s' failed to initialize. Please see attached cause exception" +
-                            ".", instance.toString() );
+                    message.append( "Component '" ).append( instanceStr ).append( "' failed to initialize" );
                 }
-                if(from == LifecycleStatus.STARTED)
+                else if ( from == LifecycleStatus.STARTED )
                 {
-                    return String.format( "Component '%s' failed to stop. Please see attached cause exception.",
-                            instance.toString()  );
+                    message.append( "Component '" ).append( instanceStr ).append( "' failed to stop" );
                 }
                 break;
             case STARTED:
-                if(from == LifecycleStatus.STOPPED)
+                if ( from == LifecycleStatus.STOPPED )
                 {
-                    return String.format( "Component '%s' was successfully initialized, but failed to start. Please " +
-                            "see attached cause exception.", instance.toString() );
+                    message.append( "Component '" ).append( instanceStr )
+                           .append( "' was successfully initialized, but failed to start" );
                 }
                 break;
             case SHUTDOWN:
-                return String.format( "Component '%s' failed to shut down. Please " +
-                        "see attached cause exception.", instance.toString() );
+                message.append( "Component '" ).append( instanceStr ).append( "' failed to shut down" );
+                break;
             default:
                 break;
         }
+        if ( message.length() == 0 )
+        {
+            message.append( "Component '" ).append( instanceStr ).append( "' failed to transition from " )
+                   .append( from.name().toLowerCase() ).append( " to " ).append( to.name().toLowerCase() );
+        }
+        message.append( '.' );
+        if ( cause != null )
+        {
+            Throwable root = rootCause( cause );
+            message.append( " Please see the attached cause exception \"" ).append( root.getMessage() ).append( "\"." );
+        }
 
-        return String.format( "Failed to transition component '%s' from %s to %s. Please see attached cause exception",
-                instance.toString(), from.name(), to.name() );
+        return message.toString();
+    }
+
+    private static Throwable rootCause( Throwable cause )
+    {
+        int i = 0; // Guard against infinite self-cause exception-loops.
+        while ( cause.getCause() != null && i++ < 100 )
+        {
+            cause = cause.getCause();
```

Comment:
Maybe it makes sense to include into error message fact that causes were cycled?